src/components/Api.js

```diff
@@ -0,0 +1,88 @@
+import Card from "./Card.js";
+import UserInfo from "./UserInfo.js";
+
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Error: ${res.status}`);
+    }
+  }
+
+  getUserInfo() {
+    return fetch("https://around-api.en.tripleten-services.com/v1/users/me", {
+      headers: this._headers,
+    }).then(this._handleResponse);
+  }
+
+  getInitialCards() {
+    return fetch("https://around-api.en.tripleten-services.com/v1/cards", {
+      headers: this._headers,
+    })
+      .then(this._handleResponse)
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  editProfile() {
+    return fetch("https://around-api.en.tripleten-services.com/v1/users/me", {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: "Marie Skłodowska Curie",
+        about: "Physicist and Chemist",
+      }),
+    });
+  }
+
+  addNewCard({ name, link }) {
+    return fetch("https://around-api.en.tripleten-services.com/v1/cards", {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({
+        name,
+        link,
+      }),
+    });
+  }
+
+  deleteCard(id) {
+    return fetch(
+      `https://around-api.en.tripleten-services.com/v1/cards/${id}`,
+      {
+        method: "DELETE",
+        headers: this._headers,
+      }
+    ).then(this._handleResponse);
+  }
+
+  addingLike(id) {
```

Comment:
Needs correction: The name of the function should begin with a verb, reflecting the action that the function performs. Next, the elements to which the action refers are described. `addLike` would be good.